relay-server/src/actors/project.rs

```diff
@@ -606,24 +607,92 @@ impl Project {
         });
     }
 
+    fn ratelimit_and_merge_buckets(
+        &self,
+        project_state: Arc<ProjectState>,
+        mut buckets: Vec<Bucket>,
+        aggregator: Addr<Aggregator>,
+        #[allow(unused_variables)] envelope_processor: Addr<EnvelopeProcessor>,
+        outcome_aggregator: Addr<TrackOutcome>,
+    ) {
+        let Some(scoping) = self.scoping() else {
+            // This shouldn't be possible since we check if project id is present before setting the new project state.
+            // TODO(tor): Find a way to properly represent this invariant in the type system.
+            relay_log::trace!(
+                "there is no scoping due to missing project id: dropping {} buckets",
+                buckets.len()
+            );
+            return;
+        };
+
+        // Only send if the project state is valid, otherwise drop the buckets.
+        if project_state.check_disabled(self.config.as_ref()).is_err() {
+            relay_log::trace!("project state invalid: dropping {} buckets", buckets.len());
+            return;
+        }
+
+        // Re-run feature flag checks since the project might not have been loaded when the buckets
+        // were initially ingested, or feature flags have changed in the meanwhile.
+        self.filter_metrics(&mut buckets);
```

Comment:
Check if buckets is empty here? See https://github.com/getsentry/relay/blob/75397010d18044a17759161e74475297fe9b0493/relay-server/src/actors/project.rs#L559-L563

Reply:
nice catch!